librz/core/analysis_tp.c

```diff
@@ -173,12 +182,22 @@ static void get_src_regname(RzCore *core, ut64 addr, char *regname, int size) {
 	rz_analysis_op_free(op);
 }
 
-static ut64 get_addr(Sdb *trace, const char *regname, int idx) {
+static ut64 get_addr(RzPVector *trace, const char *regname, int idx) {
 	if (!regname || !*regname) {
 		return UT64_MAX;
 	}
-	const char *query = sdb_fmt("%d.reg.read.%s", idx, regname);
-	return rz_num_math(NULL, sdb_const_get(trace, query, 0));
+	if (idx >= rz_pvector_len(trace)) {
```

Comment:
check also for `i < 0`, since `idx` is an int and not a `size_t`.

Reply:
Maybe it makes sense to use `unsigned int` specifically?

Reply:
size_t is better, if we want to change type.